.github/workflows/publish-cli-bins.yaml

```diff
@@ -20,88 +20,89 @@ jobs:
-          # Split os-arch into os and arch
-          IFS="-" read -r GOOS GOARCH <<< "${{ matrix.os_arch }}"
-          # Export variables to GITHUB_ENV for subsequent steps
-          {
-            echo "GOOS=$GOOS"
-            echo "GOARCH=$GOARCH"
-          } >> $GITHUB_ENV
-
-      - name: Checkout Code
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - name: Cache Git Repository
-        uses: actions/cache@v4
-        with:
-          path: .git
-          key: ${{ runner.os }}-git-${{ hashFiles('**/.git') }}
-          restore-keys: |
-            ${{ runner.os }}-git-
-
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
           go-version: 1.21
 
-      # Cache Go modules
-      - name: Cache Go Modules
-        uses: actions/cache@v4
+      - name: Checkout Code
+        uses: actions/checkout@v4
         with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+          fetch-depth: 0
 
       # Install dependencies
       - name: Install Dependencies
         run: go mod tidy
 
       - name: Build
         run: |
-          GOOS="${GOOS}" GOARCH="${GOARCH}" make build-cli
-          SHA=$(shasum -a 256 bin/flightctl | awk '{ print $1 }')
-          echo "${SHA}" > "flightctl-${{ matrix.os_arch }}-sha256.txt"
-          mv bin/flightctl "flightctl-${{ matrix.os_arch }}"
-          tar cvf "flightctl-${{ matrix.os_arch }}.tar.gz" "flightctl-${{ matrix.os_arch }}"
+          make multiarch-build-cli
+          pwd
+
+  save:
+    name: Save artefacts (linux)
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        include:
+          - os_arch: "linux-amd64"
+            zip: ".tar.gz"
+            ext: ""
+          - os_arch: "linux-arm64"
+            zip: ".tar.gz"
+            ext: ""
+          - os_arch: "darwin-amd64"
+            zip: ".zip"
+            ext: ""
+          - os_arch: "darwin-arm64"
+            zip: ".zip"
+            ext: ""
+          - os_arch: "windows-amd64"
+            zip: ".zip"
+            ext: ".exe"
+          - os_arch: "windows-arm64"
+            zip: ".zip"
+            ext: ".exe"
 
+    steps:
       - name: Save tar binary
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         uses: actions/upload-artifact@v4
         with:
-          name: flightctl-${{ matrix.os_arch }}.tar.gz
-          path: flightctl-${{ matrix.os_arch }}.tar.gz
+          name: "flightctl-${{ matrix.os_arch }}${{ matrix.zip }}"
+          path: "flightctl-${{ matrix.os_arch }}${{ matrix.zip }}"
 
       - name: Save binary
         uses: actions/upload-artifact@v4
         with:
-          name: flightctl-${{ matrix.os_arch }}
-          path: flightctl-${{ matrix.os_arch }}
+          name: "flightctl-${{ matrix.os_arch }}${{ matrix.ext }}"
```

Comment:
you may need to fix the shell script to also generate the non zipped binaries, I didn't account for this, sorry.